lua/dashboard/session.lua

```diff
@@ -35,33 +29,81 @@ function session.session_save(name)
   end
 
   local file_name = name == nil and project_name() or name
-  local file_path = db.session_directory .. '/' .. file_name .. '.vim'
+  local file_path = db.session_directory .. path_sep .. file_name .. '.vim'
   api.nvim_command('mksession! ' .. fn.fnameescape(file_path))
   vim.v.this_session = file_path
-  vim.notify('Session ' .. file_name .. ' is now persistent')
+
+  if db.session_verbose then
+    vim.notify('Session ' .. file_name .. ' is now persistent')
+  else
+    vim.notify('This session is now persistent')
+  end
+  db.session_cp_name = project_name()
 end
 
 function session.session_load(name)
   local file_name = name == nil and project_name() or name
-  local file_path = db.session_directory .. '/' .. file_name .. '.vim'
+  local file_path = db.session_directory .. path_sep .. file_name .. '.vim'
 
   if vim.v.this_session ~= '' and fn.exists('g:SessionLoad') == 0 then
     api.nvim_command('mksession! ' .. fn.fnameescape(vim.v.this_session))
   end
 
-  if fn.filereadable(file_path) then
+  if fn.filereadable(file_path) == 1 then
     vim.cmd([[ noautocmd silent! %bwipeout!]])
     api.nvim_command('silent! source ' .. file_path)
 
     if vim.opt.laststatus:get() == 0 then
       vim.opt.laststatus = 2
     end
 
-    vim.notify('Loaded ' .. file_path .. ' session')
+    if db.session_verbose then
+      vim.notify('Loaded ' .. file_path .. ' session')
+    else
+      vim.notify('Session loaded')
+    end
+    db.session_cp_name = project_name()
     return
   end
 
-  vim.notify('The session ' .. file_path .. ' does not exist')
+  if db.session_verbose then
```

Comment:
these code check is duplicate. you used them three times.
```lua
local msg = db.session_verbose and 'The session ' .. file_path .. ' does not exist' or 'No saved session for this directory'
vim.notify(msg)
```

Reply:
The verbose check is performed on every `vim.notify` because those are different functions with different message text. We could have a local msg variable to store the messages but I don't think it's really necessary for this since we would have to set msg to a different text when saving, loading and deleting. The "No saved session for this directory" message is the only one that will be printed in two cases because I used the same wording for session loading and deletion.

Reply:
`local session_cp_name` at the head of file is enough . it's also in mem.

Reply:
You are right... Nice. Changed it.